---
title: "Final Project"
author: "Liu Huihang"
date: "12/7/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Definition

### eQTL

Genetical genomics experiments have now been routinely conducted to measure both the genetic markers and gene expression data on the same subjects. The gene expression levels are often treated as quantitative traits and are subject to standard genetic analysis in order to identify the gene expression quantitative loci (eQTL). 


### Yeast

酵母菌一共有 16 条染色体, 全部记录在数据中. 

### Genetic Markers. 
A genotype indicates the genetic state of the DNA in an organism. 
It is a theoretical construct describing a genetic situation that explains the observed properties (phenotype, see below) of a strain. 



## Data Preparation

The **yeast** dataset is a subset of the widely studied yeast expression dataset comprising of $112$ *F1* segregants from a cross between *BY4716* and *RM11-1a* strains of Saccharomyces Cerevisiae.

The original dataset consists of expression values reported as $log_2(\text{sample} / \text{BY reference})$ for $7085$ genes. 


The data can be accessed in Gene Expression Omnibus (GEO) by accession number (GSE1990). 

```{r, message=FALSE, eval=FALSE, include=FALSE}
# Original data
# https://warwick.ac.uk/fac/sci/moac/people/students/peter_cock/r/geo/
# dat <- read.csv("GSE1990_series_matrix.txt", header=TRUE, sep = "\t")

library(Biobase)
library(GEOquery)

# gse1990 <- getGEO('GSE1990', destdir=".")
# load series and platform data from GEO

gset <- getGEO("GSE1990", GSEMatrix =TRUE, getGPL=FALSE, destdir="./Data")
if (length(gset) > 1) idx <- grep("GPL118", attr(gset, "names")) else idx <- 1
gset <- gset[[idx]]

# set parameters and draw the plot

# dev.new(width=4+dim(gset)[[2]]/5, height=6)   # unsupported in rmd
par(mar=c(2+round(max(nchar(sampleNames(gset)))/2), 4, 2, 1))
title <- paste("GSE1990", '/', annotation(gset), " selected samples", sep ='')
BiocGenerics::boxplot(exprs(gset), boxwex=0.7, notch=T, main=title, outline=FALSE, las=2, plot = TRUE)
```

Value distribution allows you to calculate and view the distribution of the values for the Samples you have selected. 
Values are the original submitter-supplied data upon which GEO2R calculations are performed. 
Viewing the distribution is important for determining if your selected Samples are suitable for comparison; see Limitations and caveats for more information. 
Generally, median-centered values are indicative that the data are normalized and cross-comparable. 

### Data Preprocessing

In **SOFAR**, the data is finally reduced to the size of $n=112, p=605, q=54$. 

In **geneNetBP**, $n=112, p=12, q=38$. 
Thus the final dataset is a data frame of $112$ observations (genotype) of $12$ variables (SNP markers) and normalized gene expression of $38$ variables (genes). 
The gene expression values are discretized around the median and have two states, $1$ (above or equal to median) and $-1$ (below median). 
There are two genotype states: $1$ or $2$. 

```{r, warning=FALSE, message=FALSE, eval=FALSE, include=FALSE}
# Data: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE1990
# geneNetBP: https://cran.r-project.org/web/packages/geneNetBP/geneNetBP.pdf
# trigger: https://bioconductor.org/packages/release/bioc/vignettes/trigger/inst/doc/trigger.pdf

library(geneNetBP)    # for data(yeast)

data(yeast)

head(yeast)

# fit.dbn(yeast[,1:12],yeast[,13:50])
```

In **trigger**, *marker*: A $3244$ × $112$ genotype matrix; *exp*: A $6216$ × $112$ gene expression matrix



```{r, message=FALSE, eval=FALSE, include=FALSE}
rm(list=ls())
library(trigger)

data(yeast)

# head(yeast$marker)
# 
# head(yeast$exp)

#reduce data size for vignette run time
set.seed(123)
#select subset of 400 traits
gidx = sort(sample(1:6216, size = 400))
yeast$exp = yeast$exp[gidx,]
yeast$exp.pos = yeast$exp.pos[gidx,]
#select subset of markers
midx = sort(sample(1:3244, size = 500))
yeast$marker = yeast$marker[midx,]
yeast$marker.pos = yeast$marker.pos[midx,]

attach(yeast)
dim(exp)


trig.obj <- trigger.build(marker=marker, exp=exp, marker.pos=marker.pos, exp.pos=exp.pos)
trig.obj
detach(yeast)


trig.obj = trigger.link(trig.obj, norm = TRUE)
plot(trig.obj, type = "link", cutoff = 1e-5)
```

```{r, message=FALSE}
rm(list=ls())
library(trigger)
library(stats)

data(yeast)

marker <- yeast$marker
res <- stats::kmeans(marker, centers=585)
idxs_ii <- list()
for (ii in 1:585){
  idxs_ii <- c(idxs_ii, list(marker[which(res$cluster == ii),]))
}

a <- idxs_ii[[1]]
b <- a[1,] - a
```