---
title: "Final Project"
author: "Liu Huihang and Xu Tianchen"
date: "12/7/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Introduction & Definition

### eQTL

Genetical genomics experiments have now been routinely conducted to measure both the genetic markers and gene expression data on the same subjects. The gene expression levels are often treated as quantitative traits and are subject to standard genetic analysis in order to identify the gene expression quantitative loci (eQTL). 


### Yeast


### Genetic Markers. 
A genotype indicates the genetic state of the DNA in an organism. 
It is a theoretical construct describing a genetic situation that explains the observed properties (phenotype, see below) of a strain. 



## Data Preparation

### Loading the data
The **yeast** dataset is a subset of the widely studied yeast expression dataset comprising of $112$ *F1* segregants from a cross between *BY4716* and *RM11-1a* strains of Saccharomyces Cerevisiae.

The original dataset consists of expression values reported as $log_2(\text{sample} / \text{BY reference})$ for $7085$ genes. 


The data can be accessed in Gene Expression Omnibus (GEO) by accession number (GSE1990). 

Value distribution allows you to calculate and view the distribution of the values for the Samples you have selected. 
Values are the original submitter-supplied data upon which GEO2R calculations are performed. 
Viewing the distribution is important for determining if your selected Samples are suitable for comparison; see Limitations and caveats for more information. 
Generally, median-centered values are indicative that the data are normalized and cross-comparable. 

```{r, message=FALSE}
rm(list=ls())
DEBUG <- FALSE

# load data from trigger package or tmp.RData
if (FALSE){
  library(trigger)
  data(yeast)
} else{
  load('tmp.RData')
}

library(stats)

marker <- yeast$marker
exp <- yeast$exp
```

### Data Preprocessing

In **SOFAR**, the data is finally reduced to the size of $n=112, p=605, q=54$. 

The gene expression values are discretized around the median and have two states, $1$ (above or equal to median) and $-1$ (below median). 
There are two genotype states: $1$ or $2$. 


In **trigger**, *marker*: A $3244$ × $112$ genotype matrix; *exp*: A $6216$ × $112$ gene expression matrix

```{r, message=FALSE}
rm(list=ls())
DEBUG <- FALSE

# load data from trigger package or tmp.RData
if (TRUE){
  library(trigger)
  data(yeast)
} else{
  load('tmp.RData')
}

library(stats)

marker <- yeast$marker
exp <- yeast$exp
```

Distance clustering, d <= 1

```{r}
# hclust
out.dist <- dist(marker, method="manhattan")
hc1.complete <- hclust(dist(marker, method="manhattan"), method="complete") # complete method using the max distance within a group
out.id <- cutree(hc1.complete, h=1)                                         # h is the maximum distance to be cut
if (DEBUG) print(max(out.id))
if (DEBUG){
  for (ii in 1:max(out.id)) {
    tmp <- marker[which(out.id==ii),]                                         
    if (is.matrix(tmp)) {
      d <- matrix(rep(1,dim(tmp)[1]),ncol=1) %*% tmp[1,] - tmp
      d.col <- colSums(abs(d))
      if (length(which(d.col != 0)) > 1){
        cat("some thing wrong at ", ii, "ther are", length(which(d.col != 0)), "diffs \n")
      }
    }
  }
}
```

We combined the markers into `r max(out.id)` blocks where the markers within a block differed by at most $1$ sample. 
We select the most repeated marker from each block as the representative marker. 

```{r}
count <- numeric(max(out.id))
for (ii in 1:max(out.id)) {
  tmp <- marker[which(out.id==ii),]
  if (!is.matrix(tmp)) {
    count[ii] <- which(out.id == ii)
    next
  }
  tmp.unique <- unique(tmp)
  if (dim(tmp.unique)[1] == 1){
    count[ii] <- which(out.id == ii)[1]
    next
  }
  tmp.count <- numeric(2)
  tmp.rec <- numeric(length(which(out.id == ii)))
  for (jj in 1:length(which(out.id == ii))) {
    want <- marker[which(out.id == ii)[jj],]
    tmp.idx <- ifelse(sum(abs(want - tmp.unique[2,])), 1, 2) 
    tmp.count[tmp.idx] <- tmp.count[tmp.idx] + 1
    tmp.rec[jj] <- tmp.idx
  }
  tmp.mostfreq <- which.max(table(tmp.rec))[1]
  count[ii] <- which(out.id == ii)[which(tmp.rec == tmp.mostfreq)[1]]
}

X <- t(marker[count,])
```

Next, I am going to reduce the dimension of $Y$. 
```{r}
# From The yeast MAPK pathway from the KEGG database http://www.genome.jp/kegg/pathway/sce/sce04011.html
MAPK.pathway.name <- c('MATALPHA1', 'MATALPHA2', 'STE2', 'STE3', 'GPA1', 'STE4', 'STE18', 'CDC42', 'CDC24', 'BEM1', 'STE20', 'BNI1', 'STE11', 'STE5', 'STE7', 'FUS3', 'MSG5', 'STE12', 'DIG1', 'DIG2', 'FAR1', 'MCM1', 'FUS1', 'WSC4', 'WSC2', 'WSC3', 'MID2', 'RHO1', 'FKS1', 'PKC1', 'BCK1', 'MKK1', 'MKK2', 'SLT2', 'MLP2', 'RLM1', 'SWI4', 'SWI6', 'FKS3', 'SHO1', 'SLN1', 'YPD1', 'SSK1', 'SSK2', 'PBS2', 'HOG1', 'MSN2', 'MSN4', 'GLO1', 'CTT1', 'RAS2', 'KSS1', 'TEC1')
# CHANGE: WSC1->WSC4; FKS2->FKS3

idx.exp <- sapply(MAPK.pathway.name, function(x) which(rownames(exp) == x))
Y <- t(exp[idx.exp,])

if (!DEBUG){
  rm("idx.exp", "MAPK.pathway.name")
}
if (!DEBUG){
  rm("count", "ii", "jj", "out.dist", "out.id", "tmp.count", "tmp.idx", "tmp.mostfreq", "tmp.rec", "want", "hc1.complete", "tmp", "tmp.unique")
}
```

Finally, we obtain the processed data:  $X \in \mathbb{R}^{112\times 949}$, $Y \in \mathbb{R}^{112\times 53}$. 



## Statistical Analysis

Using some statistical methods to analysis the data. 

Repeat the simulation study of SOFAR. 

### Simple data analysis:

1. Estimate the covariance matrix of X

2. The marginal correlation between $X_i$ and $Y_j$

We can apply a marginal gene-marker association analysis to identify markers that are associated with the expression levels of at least two genes
with p-value less than a specific level such as $0.05$. 
```{r}
rm(list=ls())
DEBUG <- FALSE
load('tmp.RData')

X.sis <- scale(X, center = TRUE, scale = TRUE)
Y.sis <- scale(Y, center = TRUE, scale = TRUE)
sis1 <- t(X.sis) %*% Y.sis[,1]
```


### Single response variable

We can use lasso to fit each cols of Y by X.

```{r, warning=FALSE, message=FALSE}
rm(list=ls())
DEBUG <- FALSE
load('tmp.RData')
library(ggplot2)
library(ggpubr)     # for ggarrange

# lasso
beta.hat <- matrix(0., dim(X)[2], dim(Y)[2])
for (ii in 1:dim(Y)[2]) {
  fit.cv <- glmnet::cv.glmnet(X, Y[,ii])
  beta.hat[, ii] <- - as.matrix(glmnet::glmnet(X, Y[,ii], lambda=fit.cv$lambda.min)$beta)
}

# heatmap
gene_names <- colnames(Y)
sample_num <- 1:112
df_lasso <- expand.grid(gene_names, sample_num)
df_lasso$fill1 <- as.vector(t(Y))
df_lasso$fill2 <- as.vector(t(X %*% beta.hat))
p1_real <-  ggplot(df_lasso, aes(Var1, Var2, fill = fill1)) +
  geom_tile() + scale_fill_gradient2(low = "red", high = "blue") + 
  theme(panel.background = element_rect(fill = 'white', colour = 'white'),
    plot.title = element_text(hjust = 0.5),
    legend.position = "none",
    axis.text.x = element_text(
      angle = 90,
      vjust = 0,
      size = 4.5,
      hjust = .5),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) + 
  ggtitle("Real Y")
p2_lasso <- ggplot(df_lasso, aes(Var1, Var2, fill = fill2)) +
  geom_tile() + scale_fill_gradient2(low = "red", high = "blue") + 
  theme(panel.background = element_rect(fill = 'white', colour = 'white'),
    plot.title = element_text(hjust = 0.5),
    legend.position = "none",
    axis.text.x = element_text(
      angle = 90,
      vjust = 0,
      size = 4.5,
      hjust = .5),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) + 
  ggtitle("Estimated by Lasso")
p_lasso <- ggarrange(p1_real, p2_lasso, nrow = 1)
p_lasso
```

The results shows that, lasso can give a sparse estimation. 
But the rank of estimated coefficient matrix $C$ is `r qr(beta.hat)$rank`, it means that $\hat{C}$ is full rank. 
Also, heatmap shows that lasso loses many a detials of $Y$. 
There are `r length(which(rowSums(abs(beta.hat)) != 0))` nonzero rows in $\hat{C}$. 
Of couse, this method does not satisfy the requirement to explain the correlation between $Y$. 
Because it just consider each $Y_j$ at each step. 

### Multi response variable

```{r, warning=FALSE}
# load the packages and codes
library(rrpack)

# Estimation
res.sofar <- sofar(Y, X, nrank=3, ic.type="GIC", control=list(lamA=0.15, lamB=0.1, lamD=0.1))

# check for sparsity of U
u.rowsums <- rowSums(abs(res.sofar$U))
nonzero.u.hat <- length(which(u.rowsums != 0))
# check for sparsity of V
v.rowsums <- rowSums(abs(res.sofar$V))
nonzero.v.hat <- length(which(v.rowsums != 0))
# print the result
cat("There are", nonzero.u.hat, "non-zero rows in the estimation of U, and", 
    nonzero.v.hat, "non-zero rows in the estimation of V.")

# heatmap of Y and estimated
heatmap_plot <- function(model_fit) {
  X <- model_fit$X
  Y <- model_fit$Y
  U <- model_fit$U
  V <- model_fit$V
  D <- diag(model_fit$D)
  gene_names <- colnames(Y)
  sample_num <- 1:112
  df <- expand.grid(gene_names, sample_num)
  df$fill1 <- (as.vector(t(Y)))
  df$fill2 <- (as.vector(t(X %*% U %*% D %*% t(V))))
  #df[df==0] <- NA
  p1 <-
    ggplot(df, aes(Var1,Var2,fill=fill1)) +
    geom_tile() +
    scale_fill_gradient2(low = "red",
                         high = "blue"
                        ) + 
    theme(panel.background = element_rect(fill = 'white', colour = 'white'), 
          plot.title = element_text(hjust = 0.5), legend.position = "none", 
          axis.text.x = element_text(
                           angle = 90,
                           vjust = 0,
                           size = 4.5,
                           hjust = .5
                         ),
          axis.title.x=element_blank(), axis.title.y=element_blank()) + 
    ggtitle("Real Y")
p2 <-
  ggplot(df, aes(Var1, Var2, fill = fill2)) + geom_tile() +
  scale_fill_gradient2(low = "red",
                       high = "blue",
                       mid = "white") + 
  theme(panel.background = element_rect(fill = 'white', colour = 'white'),
        plot.title = element_text(hjust = 0.5),legend.position = "none", 
        axis.text.x = element_text(
                         angle = 90,
                         vjust = 1,
                         size = 4.5,
                         hjust = .5
                       ),
        axis.title.x=element_blank(),axis.title.y=element_blank()) +
  ggtitle("Estimated Y")
p <- ggarrange(p1, p2, nrow = 1)
print(p)
}

heatmap_plot(res.sofar)

# plot sofar
svd_plot <- function(fit_model) {
  U <- fit_model$U
  V <- fit_model$V
  D <- fit_model$D
  X <- fit_model$X
  Y <- fit_model$Y
  n <- nrow(X)
  rank <- length(D)
  # put 3 sets of data together into a col
  plot_x <- plot_y <-   matrix(rep(0, rank * n), ncol = 1)
  plot_label <- character(rank * n)
  for (i in 1:rank) {
    plot_x[((i - 1) * n + 1):(i * n)] <- X %*% U[, i]
    plot_y[((i - 1) * n + 1):(i * n)] <- Y %*% V[, i]
    plot_label[((i - 1) * n + 1):(i * n)] <- paste("Layer", i, sep = " ")
  }
  df <- as.data.frame(cbind(plot_x, plot_y, plot_label))
  p <- ggplot(df, aes(x = plot_x, y = plot_y)) + 
    geom_point() + geom_smooth(method = 'lm', se = FALSE) +
    facet_wrap( ~ plot_label, nrow = 1) +
    xlab("Latent Predictor") + ylab("Latent Response")
  print(p)
}

svd_plot(res.sofar)
```


Further, we performed a marginal gene-marker association analysis to identify markers that are associated with the expression levels of at least two genes with a p-value less than $0.05$, resulting in a total of $p = 776$ markers.

```{r, warning=FALSE}
idx.marker <- numeric(dim(X)[2])
for (ii in 1:dim(X)[2]){
  counter.ii <- 0
  for (jj in 1:dim(Y)[2]) {
    if (cor.test(X[,ii], Y[,jj])$p.value < 0.05) counter.ii <- counter.ii + 1
    if (counter.ii == 2) {
      idx.marker[ii] <- 1
      break
    }
  }
}
X <- X[, which(idx.marker == 1)]

# Estimation
res.sofar <- sofar(Y, X, nrank=3, ic.type="GIC", control=list(lamA=0.2, lamB=0.1, lamD=0.1))

# check for sparsity of U
u.rowsums <- rowSums(abs(res.sofar$U))
nonzero.u.hat <- length(which(u.rowsums != 0))
# check for sparsity of V
v.rowsums <- rowSums(abs(res.sofar$V))
nonzero.v.hat <- length(which(v.rowsums != 0))
# print the result
cat("There are", nonzero.u.hat, "non-zero rows in the estimation of U, and", 
    nonzero.v.hat, "non-zero rows in the estimation of V.")

# # heatmap of Y
heatmap_plot(res.sofar)

# plot sofar
svd_plot(res.sofar)
```